src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,17 +1,68 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { useDispatch } from 'react-redux';
+import { useAppSelector } from '../../app/hooks';
+import { actions as actionsFilter, Filter } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const dispatch = useDispatch();
+  const { filter, query } = useAppSelector(state => state.filter);
+  const todos = useAppSelector(state => state.todos);
+
+  const filterFunc = () => {
+    const allTodos = [...todos];
+
+    const filterTodos = allTodos.filter(todo => {
+      switch (filter) {
+        case Filter.ALL:
+          return todo.title.toLowerCase().includes(query.toLowerCase());
+
+        case Filter.ACTIVE:
+          return !todo.completed && todo.title.toLowerCase()
```

Comment:
forget to write it the first time

you might create a variable `const isMathQuery = todo.title.toLowerCase().includes(query.toLowerCase())` above the switch block and reuse it in every case. Just FYI